src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -803,4 +803,21 @@ public void scheduleModelingExercises(Exam exam) {
         exam.getExerciseGroups().stream().flatMap(group -> group.getExercises().stream()).filter(exercise -> exercise instanceof ModelingExercise).map(Exercise::getId)
                 .forEach(instanceMessageSendService::sendModelingExerciseSchedule);
     }
+
+    /**
+     * To avoid direct access to the {@link ExamRepository}, we use delegation to find all current and upcoming exams.
+     * @return List<Exam>
+     */
+    public List<Exam> findAllCurrentAndUpcomingExams() {
+        return examRepository.findAllCurrentAndUpcomingExams();
+    }
+
+    /**
+     * To avoid direct access to the {@link ExamRepository}, we use delegation to find the {@link Exam} based on its id or esle throw an exception.
+     * @param examId the id ofg the exam
+     * @return Exam
+     */
+    public Exam findByIdOrElseThrow(Long examId) {
```

Comment:
And two times here.
I really think this is not needed, as it already introduces a bunch of useless methods in services that are already very large (800+ lines in this case)